cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -254,13 +254,10 @@ func (p *dockerProject) Build(
 			}
 
 			_, err := os.Stat(path)
-			if err != nil && !errors.Is(err, os.ErrNotExist) {
-				task.SetError(fmt.Errorf("reading dockerfile: %w", err))
-				return
-			}
-
-			if errors.Is(err, os.ErrNotExist) {
-				// Build the container from source
+			if errors.Is(err, os.ErrNotExist) && serviceConfig.Docker.Path == "" {
```

Comment:
Could you also write tests for this?

Reply:
The success case here is covered by the test [here](https://github.com/Azure/azure-dev/blob/311d3c832cbe7f6392a632f8949b020dbdd125a6/cli/azd/pkg/project/framework_service_docker_test.go#L340). The failure case is an `err != nil { return err }` so I think we're OK not testing explicitly for the failure.

Reply:
It looks like that tests when there is no docker file, not when it has an invalid docker file.  Is that true?

Reply:
It indeed tests that no `docker.path` is specified. Are you asking to the failure case of "docker file not found", i.e. the `err != nil { return err }` case?